Library/Formula/iojs.rb

```diff
@@ -0,0 +1,125 @@
+class Iojs < Formula
+  homepage "https://iojs.org/"
+  url "https://iojs.org/dist/v1.0.2/iojs-v1.0.2.tar.gz"
+  sha256 "39fa602bf8dda874682d9c0380311de3295997c0b674b5e8aec0b988912cd9d1"
+
+  option "with-debug", "Build with debugger hooks"
+  option "with-completion", "install bash completion for npm"
+
+  depends_on :python => :build
+
+  resource "npm" do
+    url "https://registry.npmjs.org/npm/-/npm-2.2.0.tgz"
+    sha1 "e9a1c4971558019f3d14f7a33aa7a7492bc195ed"
+  end
+
+  def install
+    args = %W[--prefix=#{libexec}/iojs --without-npm]
+    args << "--debug" if build.with? "debug"
+
+    system "./configure", *args
+    system "make", "install"
+    (bin+"iojs").write iojs_script
+
+    resource("npm").stage buildpath/"npm_install"
+
+    # make sure npm can find iojs
+    ENV.prepend_path "PATH", libexec/"iojs/bin"
+    # set log level temporarily for npm's `make install`
+    ENV["NPM_CONFIG_LOGLEVEL"] = "verbose"
+
+    cd buildpath/"npm_install" do
+      system "./configure", "--prefix=#{libexec}/npm"
+      system "make", "install"
+    end
+    (bin+"npm-iojs").write npm_script
+
+    if build.with? "completion"
+      bash_completion.install \
+        buildpath/"npm_install/lib/utils/completion.sh" => "npm"
+    end
+  end
+
+  # This tells iojs where to find npm modules, and executes the hidden iojs.
+  def iojs_script
+    <<-EOS.undent
+      #!/usr/bin/env bash
+      export NODE_PATH=#{HOMEBREW_PREFIX}/share/iojs/
+      exec #{libexec}/iojs/bin/iojs "$@"
+    EOS
+  end
+
+  # This tells npm where to find iojs.
+  def npm_script
+    <<-EOS.undent
+      #!/usr/bin/env bash
+      export PATH=#{libexec}/iojs/bin:$PATH
+      exec #{HOMEBREW_PREFIX}/share/iojs/bin/npm "$@"
+    EOS
+  end
+
+  def post_install
+    iojs_bindir = HOMEBREW_PREFIX/"share/iojs/bin"
+    iojs_bindir.mkpath
+    iojs_modules = HOMEBREW_PREFIX/"share/iojs/lib/node_modules"
+    iojs_modules.mkpath
+    npm_exec = iojs_modules/"npm/bin/npm-cli.js"
+    # Kill npm but preserve all other modules across iojs updates/upgrades.
+    rm_rf iojs_modules/"npm"
+
+    cp_r libexec/"npm/lib/node_modules/npm", iojs_modules
+    ln_sf npm_exec, iojs_bindir/"npm"
+
+    # Let's do the manpage dance. It's just a jump to the left.
+    # And then a step to the right, with your hand on rm_f.
+    ["man1", "man3", "man5", "man7"].each do |man|
+      mkdir_p HOMEBREW_PREFIX/"share/man/#{man}"
+      rm_f Dir[HOMEBREW_PREFIX/"share/man/#{man}/{npm.,npm-,npmrc.}*"]
+      Dir[libexec/"npm/share/man/#{man}/npm*"].each { |f| ln_sf f, HOMEBREW_PREFIX/"share/man/#{man}" }
```

Comment:
Nice spot! Will fix as soon as my internet decides it wants to go a little faster than 7KB/s :crying_cat_face: